control-plane/subcommand/sync-catalog/command.go

```diff
@@ -462,64 +458,15 @@ func (c *Command) Run(args []string) int {
 
 // remove all k8s services from Consul.
 func (c *Command) removeAllK8SServicesFromConsulNode(consulClient *api.Client) error {
-	node, _, err := consulClient.Catalog().NodeServiceList(c.flagPurgeK8SServicesFromNode, &api.QueryOptions{Filter: c.flagFilter})
+	_, err := consulClient.Catalog().Deregister(&api.CatalogDeregistration{
```

Comment:
The concern for doing the de-registration in one shot is that JPMC reported that they have thousands of services and they hope the de-registration would not impact the consul performance, that's the context of implementing batch purging in the implementation.

Reply:
so deregistering the node shouldn't have that effect, we can run some perf tests to check on this though